libsql/src/hrana/stream.rs

```diff
@@ -0,0 +1,372 @@
+use crate::hrana::pipeline::{
+    BatchStreamReq, ClientMsg, CloseSqlStreamReq, DescribeStreamReq, ExecuteStreamReq, Response,
+    SequenceStreamReq, ServerMsg, StoreSqlStreamReq, StreamRequest, StreamResponse,
+    StreamResponseOk,
+};
+use crate::hrana::proto::{Batch, BatchResult, DescribeResult, Stmt, StmtResult};
+use crate::hrana::{HranaError, HttpSend, Result};
+use futures::lock::Mutex;
```

Comment:
We should use the tokio mutex if we need an async mutex and not the one inside futures crate. Its a bit unsafe and not as fully tested as the tokio one which I trust much more.